### PR TITLE
Jetpack Cloud: track page views in Jetpack Cloud

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Track page views in every Jetpack Cloud page.

#### Questions

* Should we track this page as well?
https://github.com/Automattic/wp-calypso/blob/master/client/landing/jetpack-cloud/sections/backups/rewind-flow/index.tsx#L72
* Should we track the Scan UpSell page? 
https://github.com/Automattic/wp-calypso/blob/master/client/landing/jetpack-cloud/sections/scan/upsell.jsx#L26

#### Testing instructions

* Go to http://jetpack.cloud.localhost:3000/
* Select a site that has Backup and Scan
* Go to Backup > Status
* Verify the following action was dispatched
```
{
  type: 'ANALYTICS_PAGE_VIEW_RECORD',
  payload: {
    service: 'default',
    url: '/backups/:site',
    title: 'Backups',
    site_type: 'jetpack',
    site: <YOUR_SITE_SLUG>
  }
}
```
* Select any Backup details
* Verify the following action was dispatched
```
{
  type: 'ANALYTICS_PAGE_VIEW_RECORD',
  payload: {
    service: 'default',
    url: '/backups/:site/detail/:backup_id',
    title: 'Backup Details',
    site_type: 'jetpack',
    site: <YOUR_SITE_SLUG>,
    backup_id: <YOUR_BACKUP_ID>
  }
}
```
* Go to Backup > Activity Log
* Verify the following action was dispatched
```
{
  type: 'ANALYTICS_PAGE_VIEW_RECORD',
  payload: {
    service: 'default',
    url: '/backups/activity/:site',
    title: 'Activity log',
    site_type: 'jetpack',
    site: <YOUR_SITE_SLUG>
  }
}
```
* Go to Scan > Scanner
* Verify the following action was dispatched
```
{
  type: 'ANALYTICS_PAGE_VIEW_RECORD',
  payload: {
    service: 'default',
    url: '/scan/:site',
    title: 'Scanner',
    site: <YOUR_SITE_SLUG>,
    site_type: 'jetpack'
  }
}
```
* Go to Scan > History
* Verify the following action was dispatched
```
{
  type: 'ANALYTICS_PAGE_VIEW_RECORD',
  payload: {
    service: 'default',
    url: '/scan/history/:site',
    title: 'History',
    site: <YOUR_SITE_SLUG>,
    site_type: 'jetpack'
  }
}
```
* Go to Settings
* Verify the following action was dispatched
```
{
  type: 'ANALYTICS_PAGE_VIEW_RECORD',
  payload: {
    service: 'default',
    url: '/settings/:site',
    title: 'Settings',
    site: <YOUR_SITE_SLUG>,
    site_type: 'jetpack'
  }
}
```

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/scan/history/index.jsx [hunk 1]
```diff
@@ -67,9 +68,10 @@ class ScanHistoryPage extends Component {
 		const { value: filter } = this.getCurrentFilter();
 		return (
 			<Main className="history">
-				<QueryJetpackScanHistory siteId={ this.props.siteId } />
 				<DocumentHead title={ translate( 'History' ) } />
 				<SidebarNavigation />
+				<QueryJetpackScanHistory siteId={ this.props.siteId } />
+				<PageViewTracker path="/scan/history/:site" title="History" />
```
Comment (enejb) [inline client/landing/jetpack-cloud/sections/scan/history/index.jsx:74]: Should we update it to Scan History?
```suggestion
				<PageViewTracker path="/scan/history/:site" title="Scan History" />
```

File: client/landing/jetpack-cloud/sections/scan/upsell.jsx [hunk 2]
```diff
@@ -22,6 +23,7 @@ function ScanUpsellPage( props ) {
 		<Main className="scan__main">
 			<DocumentHead title="Scanner" />
 			<SidebarNavigation />
+			<PageViewTracker path="/scan/:site" title="Scanner UpSell" />
```
Comment (enejb) [inline client/landing/jetpack-cloud/sections/scan/upsell.jsx:26]: Lower case S. 
```suggestion
			<PageViewTracker path="/scan/:site" title="Scanner Upsell" />
```

**Comment by rcanepa** [comment]
I pushed a new update because I forgot to add the `properties` prop to all instances of the PageViewTracker component. Now they should look like this:

```
<PageViewTracker
  path="/backups/activity/:site"
  title="Activity log"
  properties={ { site: siteSlug } }
/>
```

**Comment by enejb** [comment]
I am seeing a JS error when I visit the /scan page and /scan/history page
See https://jetpack-hash-33d87cde52f24cf4511974ccf2e52742b70fc8f5.calypso.live/scan/history/

And the the page fails to load. I think it has to do with { this.props.siteSlug }.

**Comment by rcanepa** [comment]
That's odd. I just checked in the link you gave me and I didn't see any error. The action includes the site's slug in the `site` property.

image

**Comment by enejb** [comment]
yeah the weird part it works for some sites but not others. 
The site that I am having trouble with is enej.wpsandbox.me/aaa

**Comment by enejb** [comment]
@rcanepa I think it is fixed now. 

Lets ship this!

**Comment by enejb** [comment]
I was looking at how the `<PageViewTracker` component was implemeneted in most of calypso and most of the sections don't include a "site" prop? 

Should we remove the siteSlug?

**Comment by rcanepa** [comment]
> @rcanepa I think it is fixed now.

Thanks for the fix!

> I was looking at how the <PageViewTracker component was implemeneted in most of calypso and most of the sections don't include a "site" prop?
>
> Should we remove the siteSlug?

Yeah. It seems that if you write `:site` on the path, it will automatically fill it for you, so you only have to worry about the rest of the parameters.

I'll remove those siteSlugs.

**Comment by rcanepa** [comment]
@enejb , PR updated. I removed those `siteSlug`s from all call sites.

**Pull request merged**